src/aioftp/client.py

```diff
@@ -769,60 +915,73 @@ def list(self, path="", *, recursive=False, raw_command=None):
 
             >>> stats = await client.list()
         """
-
-        class AsyncLister(AsyncListerMixin):
-            stream = None
-
-            async def _new_stream(cls, local_path):
-                cls.path = local_path
-                cls.parse_line = self.parse_mlsx_line
+        ctx = self
+
+        class AsyncLister(
+            AsyncIterator[Tuple[pathlib.PurePath, InfoDict]],
+            AsyncListerMixin[Tuple[pathlib.PurePath, InfoDict]],
+        ):
+            stream: Optional[DataConnectionThrottleStreamIO] = None
+            directories: Deque[Tuple[_Path, InfoDict]]
+            parse_line: Callable[[bytes], Tuple[pathlib.PurePath, InfoDict]]
+
+            async def _new_stream(
+                self,
+                local_path: _Path,
+            ) -> Optional[DataConnectionThrottleStreamIO]:
+                self.path = local_path
+                self.parse_line = ctx.parse_mlsx_line
                 if raw_command not in [None, "MLSD", "LIST"]:
                     raise ValueError(
-                        "raw_command must be one of MLSD or " f"LIST, but got {raw_command}",
+                        f"raw_command must be one of MLSD or LIST, but got {raw_command}",
                     )
                 if raw_command in [None, "MLSD"]:
                     try:
-                        command = ("MLSD " + str(cls.path)).strip()
-                        return await self.get_stream(command, "1xx")
+                        command = ("MLSD " + str(self.path)).strip()
+                        return await ctx.get_stream(command, Code("1xx"))
                     except errors.StatusCodeError as e:
                         code = e.received_codes[-1]
                         if not code.matches("50x") or raw_command is not None:
                             raise
                 if raw_command in [None, "LIST"]:
-                    cls.parse_line = self.parse_list_line
-                    command = ("LIST " + str(cls.path)).strip()
-                    return await self.get_stream(command, "1xx")
+                    self.parse_line = ctx.parse_list_line
+                    command = ("LIST " + str(self.path)).strip()
+                    return await ctx.get_stream(command, Code("1xx"))
+                return None
+
+            @override
```

Comment:
Don't get why it should be overridden.

Reply:
Because AsyncLister is an iterator. Inherited classes are iterators too

Reply:
Strange, I have no errors with code below:
```py
from typing import AsyncIterator, AsyncIterable, Awaitable, TypeVar, Generator, Self

T = TypeVar("T")

class AsyncListerMixin(AsyncIterable[T]):
    async def _to_list(self) -> list[T]:
        items = []
        async for item in self:
            items.append(item)
        return items

    def __await__(self) -> Generator[None, None, list[T]]:
        return self._to_list().__await__()
    
        
class C(AsyncListerMixin[int]):
    def __aiter__(self) -> Self:
        return self
        
    async def __anext__(self) -> int:
        return 1
        
async def g() -> None:
    vs1: list[int] = []
    async for v in C():
        vs1.append(v)
        
    vs2: list[int] = await C()
```

Reply:
By default, mypy and pyright do not require the use of `@overload`. To enable it, you need to specify `enable_error_code = ["explicit-override"]` and `reportImplicitOverride = true` respectively.

Reply:
Yes, `@override` is just a hint for the type system